bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunction.java

```diff
@@ -42,9 +43,11 @@ public interface QuantityTypeArithmeticGroupFunction extends GroupFunction {
     abstract class DimensionalGroupFunction implements GroupFunction {
 
         protected final Class<? extends Quantity<?>> dimension;
+        protected final @Nullable Item baseItem;
 
-        public DimensionalGroupFunction(Class<? extends Quantity<?>> dimension) {
+        public DimensionalGroupFunction(Class<? extends Quantity<?>> dimension, @Nullable Item baseItem) {
```

Comment:
I think `baseItem` shouldn't be `Nullable`. That further simplies the logic as well.

Reply:
I kept it Nullable because the JUnit tests work on that that case (which causes it to default to the first member Item). However as far as I can tell, there is no instance in actual OH run-time where a null `GroupItem::baseItem` would ever occur. So I am happy to refactor this, but note that it would require modifying existing JUnit tests, which I was reluctant to do in order to avoid self delusion. => @mherwege are you Ok with such change?

PS if we would do this, then possibly the 'dimension' argument can also be removed from the constructor. ??

Reply:
Yes, I would refactor this. And yes, you could remove the dimension argument as well.

Reply:
Done